src/fontinfo.rs

```diff
@@ -0,0 +1,940 @@
+use serde::de::Deserializer;
+use serde::ser::{SerializeSeq, Serializer};
+use serde::{Deserialize, Serialize};
+
+use crate::shared_types::Guideline;
+use crate::Error;
+
+// The specification is vague about data type limits, usually implicitly meaning
+// Python types. Since Python is dynamic, the spec does not nail down the exact type
+// in several locations and we have to assume the biggest type that can hold it.
+type Integer = i32;
+type NonNegativeInteger = u32;
+type IntegerOrFloat = f64;
+type Float = f64;
+type NonNegativeIntegerOrFloat = f64; // Must be validated to be non-negative.
```

Comment:
I would do this by making a newtype struct, `NonNegativeIntegerOrFloat(f64)`, which can only be constructed in a way that validates its input.

Reply:
Hm. Is it worth it? It's only used for unitsPerEm which is either an int or a float but must be positive. We can also remove the `type` and just check for this in `validate`.

Reply:
either way is fine.